app/src/main/java/com/eveningoutpost/dexdrip/ui/helpers/BitmapUtil.java

```diff
@@ -86,6 +56,84 @@ public static Bitmap getTiled(final Bitmap source, final int xSize, final int yS
         return tiledBitmap;
     }
 
+    public static Bitmap getBackgroundBitmap(final int xSize, final int ySize, final String background_file) {
+        Bitmap tiledBitmap = null;
+        try {
+            tiledBitmap = BitmapLoader.bitmapFromBundleCache(background_file);
+            if (tiledBitmap == null) {
+                android.util.Log.d("NumberWall", "Regenerating image");
+
+                final Uri background_uri = Uri.parse(background_file);
+
+                Bitmap image_bitmap = null;
+                if (background_file.startsWith("content://")) {
+                    FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor();
+                    image_bitmap = BitmapFactory.decodeFileDescriptor(fileDescriptor);
+                } else {
+                    File file = new File(background_file);
+                    FileInputStream in = new FileInputStream(file);
+                    image_bitmap = BitmapFactory.decodeStream(in);
+                    in.close();
+                }
+
+                Bitmap rotated_bitmap = image_bitmap;
+                final Matrix image_matrix = new Matrix();
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+                    ExifInterface exif = null;
+                    if (background_file.startsWith("content://")) {
+                        FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor(); // reset
+                        exif = new ExifInterface(fileDescriptor);
+                    } else {
+                        File file = new File(background_file);
+                        FileInputStream in = new FileInputStream(file);
```

Comment:
Please combine it with the previous line into `FileInputStream in = new FileInputStream(new File(background_file));`

Reply:
apparently I can just `new FileInputStream(background_file)` with the path for the file
![image](https://user-images.githubusercontent.com/11080704/140375851-f30158bf-b962-4686-8615-8188bfa672b8.png)